File: jcv-core/src/main/java/com/ekino/oss/jcv/core/validator/DefaultJsonValidator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.validator;

import com.ekino.oss.jcv.core.JsonContextMatcher;
import com.ekino.oss.jcv.core.JsonValidator;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.skyscreamer.jsonassert.ValueMatcher;

@Getter
@RequiredArgsConstructor
class DefaultJsonValidator<T> implements JsonValidator<T> {

    private final JsonContextMatcher contextMatcher;
    private final ValueMatcher<T> valueComparator;
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/validator/DefaultJsonValidator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.validator

import com.ekino.oss.jcv.core.JsonContextMatcher
import com.ekino.oss.jcv.core.JsonValidator
import org.skyscreamer.jsonassert.ValueMatcher

internal class DefaultJsonValidator<T>(
  private val contextMatcher: JsonContextMatcher,
  private val valueComparator: ValueMatcher<T>,
) : JsonValidator<T> {
  override fun getContextMatcher(): JsonContextMatcher = contextMatcher

  override fun getValueComparator(): ValueMatcher<T> = valueComparator
}
